server/sonar-server/src/main/java/org/sonar/server/permission/ws/GroupsAction.java

```diff
@@ -82,25 +88,62 @@ public void define(WebService.NewController context) {
 
   @Override
   public void handle(Request wsRequest, Response wsResponse) throws Exception {
+    GroupsWsRequest groupsRequest = toGroupsWsRequest(wsRequest);
+    WsGroupsResponse groupsResponse = doHandle(groupsRequest);
+    writeProtobuf(groupsResponse, wsRequest, wsResponse);
+  }
+
+  private WsGroupsResponse doHandle(GroupsWsRequest request) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00112)